adrs/0001-cli-design.md

```diff
@@ -0,0 +1,306 @@
+* get - retrieving a specific entity 
+
+Some examples with components:
+
+```commandline
+griffon entities components list --name curl
+```
+
+```commandline
+griffon entities components get --uuid
+griffon entities components get --nvr
+griffon entities components get --purl
+```
+
+```commandline
+griffon entities flaws list 
+```
+
+```commandline
+griffon entities flaws get --cve_id 
+griffon entities flaws get --uuid 
+```
+
+### Service operations
+
+Modules for process/queries operations are explicitly isolated to enable reuse in other applications.
+
+We achieve isolation by enforcing interfaces (python Protocol) which constrain definition
+ ensuring the python modules can be easily used in future contexts.
+
+#### Query operations
+
+Perform read only cross service queries answering useful questions, some examples being:
+
+* Which unfixed CVE are affecting a component?
+* Which unfixed CVE are affecting a product + version + stream?
+* Given a CVE ID, what products are affected?
+* Given a CVE ID, what components are affected?
+* What products + version + stream contain a given component (e.g. full text search)?
+* What are the fixed CVEs for a product + version + stream?
+* What are the fixed CVEs for a component?
+* What are the won’t fix CVEs for a component?
+* What are the won’t fix CVEs for a product?
+* How many CVE’s are filed against a product + version
+
+For example, to get a list of cves affecting a specific component:
+```commandline
+> griffon queries component_cves --purl "pkg:rpm/redhat/curl@7.29.0-19.el7?arch=aarch64" --affectedness AFFECTED 
+ ```
+
+A set of 'canned' queries masks the complexity of calling up individual services and synthesizing results manually.
+
+#### Process operations
+
+A **process** represents business logic typically resulting in an update of state on service entities.
+
+### Manage operations
+A separate entrypoint for management operations is provided as convenience.
+
+In the short term, this area is meant for internal usage for managing core data services.
+
+```commandline
+> griffon manage osidb status 
+> griffon manage corgi status
+```
+
+### Plugins
+
+We **MAY** develop an extensible set of plugins to external services as an 'incubation' area for 
+integrating with third party services. This gives the CLI a good 'swiss army knife' utility as
+well as help identify integrations with more query/processes leveraging 3rd party services.
+
+A few example plugins are provided, for example, an example plugin integrating with https://osv.dev:
+
+```commandline
+> griffon z_osv query-by-commit-hash --commit_hash 6879efc2c1596d11a6a6ad296f80063b558d5e0f
+
+```
+Plugins are dynamically 'registered' once they are copied into the griffon/commands/plugins folder
+allowing individual end users to develop their own customisations.
+
+## Autocompletion
+
+The python click module [provides](https://click.palletsprojects.com/en/8.1.x/shell-completion/) provides custom shell completion.
```

Comment:
nitpick: double provides

Reply:
thx - I am making changes locally will be pushing draft adr as part of init set of commits to main